brushfire-core/src/main/scala/com/stripe/brushfire/TDigest.scala

```diff
@@ -0,0 +1,78 @@
+package com.stripe.brushfire
+
+import com.tdunning.math.stats.TDigest
+import com.twitter.algebird.{Monoid, Semigroup}
+
+private [this] case object TDigestSemigroup extends Semigroup[TDigest] {
+  override def plus(l: TDigest, r: TDigest): TDigest = {
+    val td = TDigest.createDigest(math.max(l.compression(), r.compression()))
+    td.add(l)
+    td.add(r)
+    td
+  }
+
+  override def sumOption(iter: TraversableOnce[TDigest]): Option[TDigest] = {
+    iter.foldLeft(None: Option[TDigest]) {
+      case (None, el) =>
+        val td = TDigest.createDigest(el.compression())
+        td.add(el)
+        Some(td)
+
+      case (f@Some(acc), el) =>
+        acc.add(el)
+        f
+    }
+  }
+}
+
+object TDigestSplitter {
+  private def splitDists[L](q: Double)(target: L, digest: TDigest): (Map[L, Long], Map[L, Long]) = {
```

Comment:
I'll do this, would be good for performance.
On Sep 4, 2015 11:32 AM, "Avi Bryant" <notifications@github.com> wrote:

> In brushfire-core/src/main/scala/com/stripe/brushfire/TDigest.scala
> <https://github.com/stripe/brushfire/pull/68#discussion_r38781018>:
>
> > +  override def sumOption(iter: TraversableOnce[TDigest]): Option[TDigest] = {
> > +    iter.foldLeft(None: Option[TDigest]) {
> > +      case (None, el) =>
> > +        val td = TDigest.createDigest(el.compression())
> > +        td.add(el)
> > +        Some(td)
> > +
> > +      case (f@Some(acc), el) =>
> > +        acc.add(el)
> > +        f
> > +    }
> > +  }
> > +}
> > +
> > +object TDigestSplitter {
> > +  private def splitDists[L](q: Double)(target: L, digest: TDigest): (Map[L, Long], Map[L, Long]) = {
>
> nit, and not necessary for merge, but it seems slightly better to have
> this function just return (Long,Long) and do the conversion to Map[L,Long]
> separately (and reusably)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stripe/brushfire/pull/68/files#r38781018>.
>